src/lib/sticky-header/sticky-header.ts

```diff
@@ -0,0 +1,321 @@
+ * Set is as 5. Because if the DEBOUNCE_TIME is set as a too large number. The sticky effect
+ * during scroll will become vary strange and can not be scrolled smoothly.
+ */
+const DEBOUNCE_TIME: number = 5;
+
+/**
+ * Directive that marks an element as a sticky-header. Inside of a scrolling container (marked with
+ * cdkScrollable), this header will "stick" to the top of the scrolling viewport while its sticky
+ * region (see cdkStickyRegion) is in view.
+ */
+@Directive({
+  selector: '[cdkStickyHeader]',
+})
+export class CdkStickyHeader implements OnDestroy, AfterViewInit {
+
+  /**
+   * Set the sticky-header's z-index as 10 in default. Make it as an input
+   * variable to make user be able to customize the zIndex when
+   * the sticky-header's zIndex is not the largest in current page.
+   * Because if the sticky-header's zIndex is not the largest in current page,
+   * it may be sheltered by other element when being stuck.
+   */
+  @Input('cdkStickyHeaderZIndex') zIndex: number = 10;
+
+  isStuck: boolean = false;
+  /** Whether the browser support CSS sticky positioning. */
+  private _isStickyPositionSupported: boolean = true;
+
+
+  /** The element with the 'cdkStickyHeader' tag. */
+  element: HTMLElement;
+  /** The upper container element with the 'cdkStickyRegion' tag. */
+  stickyParent: HTMLElement | null;
+  /** The upper scrollable container. */
+  upperScrollableContainer: HTMLElement;
+  /**
+   * The original css of the sticky element, used to reset the sticky element
+   * when it is being unstuck
+   */
+  private _originalStyles = {} as CSSStyleDeclaration;
+  /**
+   * 'getBoundingClientRect().top' of CdkStickyRegion of current sticky header.
+   * It is used with '_stickyRegionBottomThreshold' to judge whether the current header
+   * need to be stuck.
+   */
+  private _stickyRegionTop: number;
+  /**
+   * Bottom of the sticky region offset by the height of the sticky header.
+   * Once the sticky header is scrolled to this position it will stay in place
+   * so that it will scroll naturally out of view with the rest of the sticky region.
+   */
+  private _stickyRegionBottomThreshold: number;
+
+  private _onScrollSubscription: Subscription;
+
+  private _onTouchSubscription: Subscription;
+
+  private _onResizeSubscription: Subscription;
+
+  constructor(element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentRegion: CdkStickyRegion,
+              platform: Platform) {
+    if (platform.isBrowser) {
+      this.element = element.nativeElement;
+      this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+      this.setStrategyAccordingToCompatibility();
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (!this._isStickyPositionSupported) {
+      this.stickyParent = this.parentRegion != null ?
+        this.parentRegion._elementRef.nativeElement : this.element.parentElement;
```

Comment:
:+1:  indented 2 more lines for `this.stickyParent = this.parentRegion != null ?  this.parentRegion._elementRef.nativeElement : this.element.parentElement;`